packages/wdio-sauce-service/src/service.ts

```diff
@@ -166,52 +230,57 @@ export default class SauceService {
         }
 
         const status = 'status: ' + (failures > 0 ? 'failing' : 'passing')
-        if (!global.browser.isMultiremote) {
+        if (!this.multiRemoteBrowser.isMultiremote) {
             log.info(`Update job with sessionId ${global.browser.sessionId}, ${status}`)
             return this.isUP ? this.updateUP(failures) : this.updateJob(global.browser.sessionId, failures)
         }
 
-        return Promise.all(Object.keys(this.capabilities).map((browserName) => {
-            log.info(`Update multiremote job for browser "${browserName}" and sessionId ${global.browser[browserName].sessionId}, ${status}`)
-            return this.isUP ? this.updateUP(failures) : this.updateJob(global.browser[browserName].sessionId, failures, false, browserName)
-        }))
+        if (this.capabilities) {
+            return Promise.all(Object.keys(this.capabilities).map((browserName) => {
+                const browser = global.browser[browserName as keyof BrowserObject] as BrowserObject
+                log.info(`Update multiremote job for browser "${browserName}" and sessionId ${browser.sessionId}, ${status}`)
+                return this.isUP ? this.updateUP(failures) : this.updateJob(browser.sessionId, failures, false, browserName)
+            }))
+        }
     }
 
-    onReload (oldSessionId, newSessionId) {
+    onReload (oldSessionId: string, newSessionId: string) {
         if (!this.isServiceEnabled && !this.isRDC) {
             return
         }
 
         const status = 'status: ' + (this.failures > 0 ? 'failing' : 'passing')
 
-        if (!global.browser.isMultiremote) {
+        if (!this.multiRemoteBrowser.isMultiremote) {
             log.info(`Update (reloaded) job with sessionId ${oldSessionId}, ${status}`)
             return this.updateJob(oldSessionId, this.failures, true)
         }
 
-        const browserName = global.browser.instances.filter(
-            (browserName) => global.browser[browserName].sessionId === newSessionId)[0]
+        const browserName = this.multiRemoteBrowser.instances.filter(
+            (browserName: string) => (global.browser[browserName as keyof BrowserObject] as BrowserObject).sessionId === newSessionId)[0]
         log.info(`Update (reloaded) multiremote job for browser "${browserName}" and sessionId ${oldSessionId}, ${status}`)
         return this.updateJob(oldSessionId, this.failures, true, browserName)
     }
 
-    async updateJob (sessionId, failures, calledOnReload = false, browserName) {
-        if (this.isRDC) {
+    async updateJob (sessionId: string, failures: number, calledOnReload = false, browserName?: string) {
+        if (this.api && this.isRDC) {
             await this.api.updateTest(sessionId, { passed: failures === 0 })
             this.failures = 0
             return
         }
 
         const body = this.getBody(failures, calledOnReload, browserName)
-        await this.api.updateJob(this.config.user, sessionId, body)
+        if (this.api && this.config) {
```

Comment:
It feels weird to check about this given that it will be always set. Maybe we can just ignore type checking here and have

```js
await this.api.updateJob(this.config?.user, sessionId, body as Job)
```

```suggestion
        if (this.config) {
```

Reply:
I wanted to do this as well, but since `updateJob` has the following signature: `updateJob(username: string, id: string, body: Job)` it will tell me that `this.conifg?.user` can be `undefined`, but the signature doesn't accept `undefined`, which means I would need to define some sort of a default, which would be even worse...